src/Exception/OperationFailed.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace MatthiasMullie\Scrapbook\Exception;
+
+/**
+ * @author Martin Georgiev <martin.georgiev@gmail.com>
+ * @copyright Copyright (c) 2017, Martin Georgiev. All rights reserved
+ * @license LICENSE MIT
+ */
+class OperationFailed extends Exception
+{
+    /**
+     * @var int
+     */
+    private $errorCode;
+
+    /**
+     * @var string
+     */
+    private $errorMessage;
+
+    /**
+     * Set the error (result) code for the (last) Memcached operation
+     *
+     * @param int $errorCode
+     *
+     * @return OperationFailed
+     */
+    public function setErrorCode($errorCode)
```

Comment:
I don't think we need setErrorCode, setErrorMessage, getErrorCode & getErrorMessage.
That stuff (getCode, getMessage) already exists in PHP's core extensions, better to make use of that functionality that duplicate it ourselves :)